lib/elixir/test/elixir/stream_test.exs

```diff
@@ -168,6 +168,14 @@ defmodule StreamTest do
     stream = Stream.cycle([1, 2, 3])
     assert is_function(stream)
 
+    assert_raise ArgumentError, fn ->
```

Comment:
I think we should assert on the message as well as the error since `ArgumentError` is pretty generic.